File: scripts/listener.py
#!/usr/bin/env python
import rospy
from std_msgs.msg import Int64
from study_pkg.msg import Control

def callback(msg):
    rospy.loginfo("Received: steer = %d, speed = %d", msg.steer, msg.speed)

rospy.init_node('listener')
rospy.Subscriber('my_chat_topic', Control, callback)
rospy.spin()

File: src/listener.cpp
#include <ros/ros.h>
#include <std_msgs/String.h>
#include <study_pkg/Control.h>

void topicCallback(const study_pkg::Control::ConstPtr& msg)
{
  ROS_INFO("Speed: %d / Steer: %d", msg->speed, msg->steer);
}

int main(int argc, char **argv)
{
  ros::init(argc, argv, "listener");

  ros::NodeHandle n;

  ros::Subscriber sub = n.subscribe("cpp_chatter", 1000, topicCallback);

  ros::spin();

  return 0;
}
